components/builder-depot/src/lib.rs

```diff
@@ -98,8 +98,8 @@ impl DepotUtil for config::Config {
 
     // Return a formatted string representing the filename of an archive for the given package
     // identifier pieces.
-    fn archive_path<T: Identifiable>(&self, ident: &T, target: &PackageTarget) -> PathBuf {
-        self.packages_path().join(format!(
```

Comment:
I don't understand this change.  Don't we lose the root directory by removing `self.packages_path()`?  What is the root directory of the new `PathBuf` that's returned?

Reply:
Yep, so basically we need this function to return _only_ the hartfile name as a PathBuf so it can be cleanly appended to the tempdir when a download or upload occurs. Before, this function would return the harfile name appended to a specific path which was less composable and added some dirtiness when creating that tempdir and dropping the downloaded files in. It was effectively nesting an unnecessary pathing structure inside the tempdir.